hibernate-core/src/main/java/org/hibernate/annotations/NamedEntityGraph.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * SPDX-License-Identifier: LGPL-2.1-or-later
+ * Copyright Red Hat Inc. and Hibernate Authors
+ */
+package org.hibernate.annotations;
+
+import java.lang.annotation.Repeatable;
+import java.lang.annotation.Target;
+import java.lang.annotation.Retention;
+
+import static java.lang.annotation.ElementType.ANNOTATION_TYPE;
+import static java.lang.annotation.ElementType.PACKAGE;
+import static java.lang.annotation.ElementType.TYPE;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+/**
+ * Same as {@linkplain jakarta.persistence.NamedEntityGraph}, but leveraging
+ * Hibernate's ability to specify the graph as text.
+ *
+ * @see org.hibernate.graph.GraphParser
+ *
+ * @since 7.0
+ * @author Steve Ebersole
+ */
+@Target({TYPE, PACKAGE, ANNOTATION_TYPE})
+@Retention(RUNTIME)
+@Repeatable(NamedEntityGraphs.class)
+public @interface NamedEntityGraph {
+	/**
+	 * The name used to identify the entity graph in calls to
+	 * {@linkplain org.hibernate.Session#getEntityGraph(String)}.
+	 * Entity graph names must be unique within the persistence unit.
+	 * <p/>
+	 * When applied to a root entity class, the name is optional and
+	 * defaults to the entity-name of that entity.
+	 */
+	String name() default "";
+
+	/**
+	 * The entity-name of the root of the entity {@linkplain #graph graph}.
+	 * <p/>
+	 * When applied to a root entity class, this is optional and
+	 * defaults to the entity-name of that entity.
+	 *
+	 * @apiNote Entity-name is used here rather than entity-class to allow
+	 * for dynamic models.
+	 */
+	String rootEntityName() default "";
```

Comment:
Would it be better if there was a way to specify this as part of the mini-language itself?

Reply:
It gets a little tricky.  Syntactically what is the difference between `MyEntity( someAttribute )` and `someAttribute( someSubAttribute )`?

Reply:
how about something like `MyEntity: someAttribute`

Reply:
The plan was to simply have this pass through `GraphParser`.  So if you are asking me if I want to change the `GraphParser` grammar, no not really.  

If we don't change `GraphParser`, there would need to be a pre-analysis of this text to determine if there is a =n entity "prefix".  I'm not against it, just not sure what it gains us.  But if others think its "nicer", I'm fine with it.

Reply:
I should mention, I had planned to change this a bit to allow either name or class:

```java
@interface NamedEntityGraph {
    ...
    String rootEntityName() default "";
    Class<?> rootEntityType() default void.class;
```

Mainly because I know most users will prefer to use the class reference.  I will agree that given just `#rootEntityName` your suggestion is nice.  

6 in one, half dozen in the other imo

Reply:
> how about something like `MyEntity: someAttribute`

So I did end up adding this to the parser syntax.  But in practice, for the named graphs, this is only needed for package-level definition.  For entity-level I just required that the annotation be placed on the root entity for the graph (in keeping with JPA variant) and so we always know the entity type there.


```java
@Entity
@NamedEntityGraph( graph="title, isbn")
class Book {...}
```

```java
@NamedEntityGraph( name="book-title-isbn", graph="Book: title, isbn")
package org.hibernate.test;
```

Reply:
> For entity-level I just required that the annotation be placed on the root entity for the graph (in keeping with JPA variant) and so we always know the entity type there.

Perfect!